src/ipfs.ts

```diff
@@ -68,21 +68,21 @@ export class InMemoryIPFS implements IPFS {
 
 export class CacheOnlyIPFS implements IPFS {
   private cache: ObjectCache
-  private getFromIpfs: boolean
+  private ipfsGatewayOptions?: IPFSGatewayOptions
 
-  constructor(cache: ObjectCache, getFromIpfs?: boolean) {
+  constructor(cache: ObjectCache, ipfsGatewayOptions?: IPFSGatewayOptions) {
```

Comment:
This is a breaking change, so we should release as such.

Reply:
Agreed.

I'm not well-versed with the tooling around that, but saw after a quick look that the commit title `feat!: <bla>` (note the bang) should be recognised as one and trigger a major version bump.

Reply:
Unfortunately `feat!:` doesn't actually trigger a breaking change bump in the npm semantic release plugin.
You need to add a commit footer with the text "Breaking Change:", it's a pain!

Here's an example commit https://github.com/algorandfoundation/algokit-client-generator-ts/commit/dd9d15769f7aed39a1ed8d3df47cec6c0dd22beb

You may find it easy to squash the GitHub UI and add the breaking change info.